From 52f60354b4d48a0b3c2a25ddabb10a47073bfb0c Mon Sep 17 00:00:00 2001
From: wangchong1995924 <15229716099@163.com>
Date: Wed, 10 Jul 2024 15:28:00 +0800
Subject: [PATCH] fix issue IABY7K 


diff --git a/isomaker/rpm.sh b/isomaker/rpm.sh
index a9efe4b..b2ba4de 100644
--- a/isomaker/rpm.sh
+++ b/isomaker/rpm.sh
@@ -203,7 +203,7 @@ function get_desktop_rpms()
 
 function get_everything_rpms()
 {
-    yum list --installroot="${BUILD}"/tmp --available | awk '{print $1}' | grep -E "\.noarch|\.${ARCH}" | grep -v "debuginfo" | grep -v "debugsource" > ava_every_lst
+    yum list --installroot="${BUILD}"/tmp --available | awk '{print $1}' | grep -E "\.noarch|\.${ARCH}" | grep -v "debuginfo\." | grep -v "debugsource\." > ava_every_lst
     parse_rpmlist_xml "exclude"
     cat parsed_rpmlist_exclude
     if [ -s parsed_rpmlist_exclude ];then
@@ -277,7 +277,7 @@ function everything_source_rpms_download()
 function everything_debug_rpms_download()
 {
     mkdir ${EVERY_DEBUG_DIR}
-    yum list --installroot="${BUILD}"/tmp --available | awk '{print $1}' | grep -E "debuginfo|debugsource" > ava_debug_lst
+    yum list --installroot="${BUILD}"/tmp --available | awk '{print $1}' | grep -E "debuginfo\.|debugsource\." > ava_debug_lst
     parse_rpmlist_xml "everything_debug_exclude"
     cat parsed_rpmlist_everything_debug_exclude
     if [ -s parsed_rpmlist_everything_debug_exclude ];then
-- 
2.27.0

