From 8843eab5ae826bd0229bb73a00f3041e892e81de Mon Sep 17 00:00:00 2001
From: Li Ping <1477412247@qq.com>
Date: Sat, 22 Feb 2025 02:17:52 +0000
Subject: [PATCH] Fixes boot failure caused by invalid volume IDs

---
 isomaker/iso.sh | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/isomaker/iso.sh b/isomaker/iso.sh
index 2bb52ac..3fbed91 100755
--- a/isomaker/iso.sh
+++ b/isomaker/iso.sh
@@ -351,12 +351,13 @@ function gen_devstation_iso()
     [ $? != 0  ] && return 1
     cd ${work_dir}/*/images
     DEVSTATION_LIVECD_TAR=$(ls *.iso)
-    mkdir -p iso && mount "$DEVSTATION_LIVECD_TAR" iso && cp -rf iso/* "${BUILD}"/iso && umount iso
+    volume_id=$(isoinfo -d -i "$DEVSTATION_LIVECD_TAR" | grep "Volume id" | awk '{print $NF}')
+    mkdir -p iso && mount "$DEVSTATION_LIVECD_TAR" iso && cp -rf iso/* "${BUILD}"/iso && sed -i '/menuentry/{s/-netinst//g}' "${BUILD}"/iso/EFI/BOOT/grub.cfg && umount iso && rm -rf "${BUILD}"/iso/Packages
     if [ "$ARCH" == "x86_64" ]; then
-        mkisofs -R -J -T -r -l -d -joliet-long -allow-multidot -allow-leading-dots -no-bak -V "${RELEASE_NAME}" -o "${OUTPUT_DIR}/${DEVSTATION_ISO_NAME}" -b isolinux/isolinux.bin -c isolinux/boot.cat -no-emul-boot -boot-load-size 4 -boot-info-table  -eltorito-alt-boot -e images/efiboot.img -no-emul-boot "${BUILD}"/iso
+        mkisofs -R -J -T -r -l -d -joliet-long -allow-multidot -allow-leading-dots -no-bak -V "${volume_id}" -o "${OUTPUT_DIR}/${DEVSTATION_ISO_NAME}" -b isolinux/isolinux.bin -c isolinux/boot.cat -no-emul-boot -boot-load-size 4 -boot-info-table  -eltorito-alt-boot -e images/efiboot.img -no-emul-boot "${BUILD}"/iso
         [ $? != 0 ] && return 1
     elif [ "$ARCH" == "aarch64" ] || [ "$ARCH" == "riscv64" ]; then
-        mkisofs -R -J -T -r -l -d -joliet-long -allow-multidot -allow-leading-dots -no-bak -V "${RELEASE_NAME}" -o "${OUTPUT_DIR}/${DEVSTATION_ISO_NAME}" -e images/efiboot.img -no-emul-boot "${BUILD}"/iso
+        mkisofs -R -J -T -r -l -d -joliet-long -allow-multidot -allow-leading-dots -no-bak -V "${volume_id}" -o "${OUTPUT_DIR}/${DEVSTATION_ISO_NAME}" -e images/efiboot.img -no-emul-boot "${BUILD}"/iso
         [ $? != 0 ] && return 1
     fi
     implantisomd5 "${OUTPUT_DIR}/${DEVSTATION_ISO_NAME}"
-- 
2.46.0

